### PR TITLE
Validate JQ query during ruletype creation/update

### PR DESCRIPTION
**Pull request by eleftherias**
# Summary

Check that JQ query is valid and parsable at the time the ruletype is created or updated.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: pkg/api/protobuf/go/minder/v1/validators.go [hunk 1]
```diff
@@ -232,14 +233,33 @@ func (jq *RuleType_Definition_Eval_JQComparison) Validate() error {
...
+			return fmt.Errorf("%w: jq profile accessor definition is empty", ErrInvalidRuleTypeDefinition)
+		}
+
+		q, err := gojq.Parse(jq.GetProfile().GetDef())
+		if err != nil {
+			return fmt.Errorf("%w: jq profile accessor definition is invalid: %s", ErrInvalidRuleTypeDefinition, err)
+		}
+		if _, err = gojq.Compile(q); err != nil {
+			return fmt.Errorf("%w: jq profile accessor definition is invalid: %s", ErrInvalidRuleTypeDefinition, err)
+		}
```
Comment (evankanderson) [inline pkg/api/protobuf/go/minder/v1/validators.go:261]: Should we define a `Validate` for `RuleType_Definition_Eval_JQComparison_Operator`, and then this code could be:

```golang
if jq == nil {
  return fmt.Errorf(...)
}
// Some stuff that could be a CEL validation expression
if jq.GetProfile != nil && jq.GetConstant() != nil {
  fmt.Errorf("... mutually exclusive")
} else if jq.GetProfile() == nill && jq.GetConstant() == nil {
  fmt.Errorf("... missing either component")
}
if err := jq.GetIngested().Validate(); err != nil {
  fmt.Errorf("...", err)
}
if err := jq.GetProfile().Validate() err != nil {
  fmt.Errorf("...', err)
}
```

**Review by evankanderson — APPROVED** [review]
One suggestion, but this looks like a nice improvement (and I see it actually caught one of our tests which had invalid params).

File: pkg/api/protobuf/go/minder/v1/validators.go (hunk 1 above)
Comment (eleftherias) [reply pkg/api/protobuf/go/minder/v1/validators.go]: Thanks for the suggestion, I've applied it

File: pkg/api/protobuf/go/minder/v1/validators.go [hunk 2]
```diff
@@ -232,14 +233,37 @@ func (jq *RuleType_Definition_Eval_JQComparison) Validate() error {
...
+		if err := jq.GetProfile().Validate(); err != nil {
+			return fmt.Errorf("%w: jq profile accessor is invalid: %w", ErrInvalidRuleTypeDefinition, err)
+		}
+	}
+
+	return nil
+}
+
+// Validate validates a rule type definition eval jq operator
+func (op *RuleType_Definition_Eval_JQComparison_Operator) Validate() error {
```
Comment (JAORMX) [inline pkg/api/protobuf/go/minder/v1/validators.go:256]: Let's check that `op` is not nil

**Review by JAORMX — APPROVED** [review]

**Pull request merged**